### ISSUE TITLE
ionic content shows up only after clicking on sidemenu

### ISSUE THREAD

**Issue by SandeshSarfare** [driftyco/ionic#7233]
I have an ionic App which was working fine but now It is showing little weird behaviour, When I run the app it does not show the data at first but when I click on side menu once or twice then it loads and shows the data. I have an nodejs api from which the data comes to my app and the same api has been put in aws api gateway, I noticed this behaviour since I implemented aws sdk for binding data in my APP.

What can be the root cause for the same? I can share the code but I am not sure where the issue is whether in controllers or views?

**Comment by SandeshSarfare** [comment]
I can see the response coming in time however content is not showing, I tried waiting for some time just to see if it shows the data later but It only shows when I click on side menu, else it has no content. I can see a warning Deferred long-running timer task(s) to improve scrolling smoothness. See crbug.com/574343 in console.

**Comment by daveshirman** [comment]
try surrounding the logic that modifies the page data with ngZone.

e.g.

```
import {NgZone} from '@angular/core';

constructor(private _zone: NgZone) { ... }

getSomeData().then(data => {
    this._zone.run(() => {
        //  ... modify class member data here.
    });
});

```

**Comment by jgw96** [comment]
@SandeshSarfare Hello, thanks for opening an issue with us! Could you provide a minimal example in a plunker for me to use to test with? Thanks!

**Comment by jgw96** [comment]
Hello all! As it seems it has been a while since there was any activity on this issue i will be closing it for now. Feel free to comment if you are still running into this issue. Thanks for using Ionic!

**Comment by SandeshSarfare** [comment]
@jgw96  Sorry for late response but I can not share the code as it is now using aws api with credentials, Do you have any idea why only on sliding the sidemenu in and out , the content shows on the page?

**Comment by SandeshSarfare** [comment]
@daveshirman  I have used angularJS 1x  so I can not use NgZone but I will try $scope.apply() and check

**Comment by SandeshSarfare** [comment]
I know for sure it is due to warning "Deferred long-running timer task(s) to improve scrolling smoothness." but do not know how to get rid of it

**Comment by heukirne** [comment]
I have the same issue, but I guess `_zone.run` is the solution everyone use:
https://forum.ionicframework.com/t/ionic2-with-real-time-data-not-updating-firebase/47189/13
http://blog.thoughtram.io/angular/2016/02/22/angular-2-change-detection-explained.html

**Comment by SandeshSarfare** [comment]
I solved it using Ionic Loader as the reponse was taking long time. My side menu was not getting initialized with the same time as other views hence views were not rendering without sidemenu being rendered, The idea is to show a loader on every response and hide it when it's a success, so it will render the views successfully.
Hope it helps.
